### ISSUE TITLE
Share indicator on file list

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/server#12874]
- [ ] Android @tobiasKaminsky 
- [x] iOS @marinofaggiana 
- [ ] Server @nextcloud/server-triage 

I am about to add avatar icon to shares, same like on web ui.

first is within the same server, second is via federated share:

Android:
![image](https://user-images.githubusercontent.com/5836855/49572481-33f75000-f93c-11e8-9680-d6f69d2ef59a.png)

web UI:
![image](https://user-images.githubusercontent.com/5836855/49572502-3f4a7b80-f93c-11e8-80b8-650cdaf45925.png)

Now some questions to discuss:
- why is federated sharing on web ui not done via avatar? Generation of remote avatars should be possible?
- how to handle space problem on mobile? show "shared"? Same for federated sharing? show beginning of userId?
- how to distinguish between "local" avatar and remote/federated one?

@nextcloud/designers 
@marinofaggiana

**Comment by tobiasKaminsky** [comment]
Removing the avatar is in my opinion the wrong way, as we have this currently on android.
And then you cannot distinguish if you have shared a file or if it a received share.

![2018-12-06-110448](https://user-images.githubusercontent.com/5836855/49576998-c3096580-f946-11e8-88e4-99aed3d52f5c.png)

**Comment by marinofaggiana** [comment]
iOS : the shared image status (circled red) are only for shared from me  

![img_1051](https://user-images.githubusercontent.com/8616947/49579444-86406d00-f94c-11e8-91a7-5f9cad366d52.PNG)

**Comment by tobiasKaminsky** [comment]
@marinofaggiana they then seem to be missing on the folder.
On android/web ui you can also click on these icons and the sharing view opens directly.
(that is the reason why the are always visible, as they are a shortcut to the sharing dialog).

**Comment by marinofaggiana** [comment]
Sorry I remembered badly ..... now I have look the code ...

**Comment by jancborchardt** [comment]
- If there's no avatar yet for any scenario, it's just because no one implemented it yet. We should move to avatars everywhere, not only for the share owners. (We specifically moved _to_ avatars because icons are bland and not informative @skjnldsv :)
- We don't show the text "Shared" anymore, the faces imply sharing. We should cut the share icon too when shared.
- It's not necessary to distinguish between local and federated avatars in that small list.
- It should be limited to 3 avatars, overlapped. Ideally the most recently active people on that share (if possible to get that info).

What do you think?

**Comment by skjnldsv** [comment]
> (We specifically moved to avatars because icons are bland and not informative @skjnldsv :)

So you agree we could get rid of the sharing icon then? :)

**Comment by tobiasKaminsky** [comment]
> * If there's no avatar yet for any scenario, it's just because no one implemented it yet. We should move to avatars everywhere, not only for the share owners. (We specifically moved _to_ avatars because icons are bland and not informative @skjnldsv :)

:+1: 

> * It's not necessary to distinguish between local and federated avatars in that small list.

:+1: 

![49572481-33f75000-f93c-11e8-9680-d6f69d2ef59a](https://user-images.githubusercontent.com/5836855/49632004-73817300-f9f4-11e8-9631-46b464d72998.png)

> * We don't show the text "Shared" anymore, the faces imply sharing. We should cut the share icon too when shared.
> * It should be limited to 3 avatars, overlapped. Ideally the most recently active people on that share (if possible to get that info).

This applies to shares that we create and share with e.g. 2-5 users.

![49572481-33f75000-f93c-11e8-9680-d6f69d2ef59a](https://user-images.githubusercontent.com/5836855/49632176-16d28800-f9f5-11e8-84f7-ce738000bb9d.png)

--> I am fine with these ideas.
What about the others?

**Comment by marinofaggiana** [comment]
for me is not good, too much video games or social network , I think that is not a util information

**Comment by tobiasKaminsky** [comment]
@jancborchardt what do you think?
@marinofaggiana I think these are crucial informations and an avatar icon is far better than a generic share icon, in my opinion.
Of course these screenshots look a bit crowded with avatars, but in reality you do not share this much…

**Comment by marinofaggiana** [comment]
@tobiasKaminsky @jancborchardt can you imagine this screenshot + icons on an iPhone 5  

![img_a04d5c31eeb2-1](https://user-images.githubusercontent.com/8616947/49725166-5bffef80-fc6b-11e8-8880-ef71001fe76a.jpeg)

**Comment by tobiasKaminsky** [comment]
We have this nearly the same way on web UI:

![image](https://user-images.githubusercontent.com/5836855/49725949-3247c800-fc6d-11e8-8da5-f851c84e564e.png)

I really like it as you can see directly from whom the share is…

**Comment by tobiasKaminsky** [comment]
This is my personal cloud, as you can see there are way less shares.

![image](https://user-images.githubusercontent.com/5836855/49726913-5c9a8500-fc6f-11e8-961b-64fa4aa9746f.png)

**Comment by jancborchardt** [comment]
> So you agree we could get rid of the sharing icon then? :)

@skjnldsv yes, see the second bullet point :wink: 

> We should cut the share icon too when shared.


-------------------

@tobiasKaminsky your screenshot of the app looks great! This is really how we should do it, so clean.

@marinofaggiana yes, I can imagine that. Currently you have a lot of different icons on the folders, but no real info. I can’t quickly find the pictures of our vacation together! But with the faces shown, I can simply find your face. :)

**Comment by tobiasKaminsky** [comment]
> I can’t quickly find the pictures of our vacation together!

Jan and Marino had a vacation together? *rumor* :rofl:

**Comment by marinofaggiana** [comment]
Done.

![IMG_5984](https://user-images.githubusercontent.com/8616947/64127206-fcc1ac80-cdb0-11e9-8a4b-7d33984b3a07.PNG)

**Comment by jancborchardt** [comment]
Oooh @marinofaggiana piu magnifico mi fratello!

**Comment by tobiasKaminsky** [comment]
I am not a UI expert (Jan is), but on Android we have the avatars as big as folder icons, and "share" symbol a bit smaller.

**Comment by skjnldsv** [comment]
> I am not a UI expert (Jan is), but on Android we have the avatars as big as folder icons, and "share" symbol a bit smaller.

yes, 32px for both

**Comment by jancborchardt** [comment]
@tobiasKaminsky good catch, I was looking at it a bit too closely I guess. :D

**Comment by tobiasKaminsky** [comment]
> yes, 32px for both

Not sure how it is on iOS, but on Android you cannot use absolute pixel, as it depends heavily on screen size & resolution.

**Comment by skjnldsv** [comment]
> Not sure how it is on iOS, but on Android you cannot use absolute pixel, as it depends heavily on screen size & resolution.

Yes, I was referring to how it was on the desktop! You're the experts here ;)

**Comment by szaimen** [comment]
I think this was implemented in all apps and webinterface in the meantime -> closing.

**Comment by tobiasKaminsky** [comment]
No, we still have "plain share" icon instead of showing federated avatar, etc.

**Comment by szaimen** [comment]
You are right. I only tested with internal shares.

**Comment by nickvergessen** [comment]
> No, we still have "plain share" icon instead of showing federated avatar, etc.

Showing a federated avatar means we need to wildcard allow all remote shares in the CSP, not sure if this is doable.

**Comment by tobiasKaminsky** [comment]
I am not 100% sure if I recall it correctly, but there was an idea to have an federated avatar endpoint.
So then the client/web on nc A would ask for user xy on nc B and depending on their privacy settings it would either give real image or placeholder.

Ref: https://github.com/nextcloud/server/issues/14564
